### PR TITLE
#370 - Active Learning Sidebar gives Internal Error

### LINKED ISSUE #370: Active Learning Sidebar gives Internal Error
**Describe the bug**
Entering the annotation page give Internal Error with stack trace of Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0

**To Reproduce**
1. Create a project and upload a document for the project
2. Click on Annotation
3. See error
4. Press back, then you can see the document.

**Expected behavior**
Should not throw error

**Stack Trace**
```
2018-07-03 16:49:29 ERROR [kumar] DefaultExceptionMapper - Unexpected error occurred
org.apache.wicket.WicketRuntimeException: Error attaching this container for rendering: [SidebarPanel [Component id = leftSidebar]]
	at org.apache.wicket.MarkupContainer.onBeforeRenderChildren(MarkupContainer.java:1837) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.onBeforeRender(Component.java:3916) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.internalBeforeRender(Component.java:950) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.beforeRender(Component.java:1018) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.internalPrepareForRender(Component.java:2236) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.prepareForRender(Component.java:2275) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.page.XmlPartialPageUpdate.writeComponent(XmlPartialPageUpdate.java:98) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.page.PartialPageUpdate.writeComponents(PartialPageUpdate.java:248) ~[wicket-core-7.10.0.jar!/:7.10.0] `

`Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.rangeCheck(ArrayList.java:653) ~[?:1.8.0_131]
	at java.util.ArrayList.get(ArrayList.java:429) ~[?:1.8.0_131]
	at de.tudarmstadt.ukp.inception.active.learning.sidebar.ActiveLearningSidebar.createFeatureEditor(ActiveLearningSidebar.java:449) ~[inception-active-learning-0.4.0-SNAPSHOT.jar!/:?]
	at de.tudarmstadt.ukp.inception.active.learning.sidebar.ActiveLearningSidebar.createRecommendationOperationForm(ActiveLearningSidebar.java:403) ~[inception-active-learning-0.4.0-SNAPSHOT.jar!/:?]
	at de.tudarmstadt.ukp.inception.active.learning.sidebar.ActiveLearningSidebar.<init>(ActiveLearningSidebar.java:176) ~[inception-active-learning-0.4.0-SNAPSHOT.jar!/:?]
	at de.tudarmstadt.ukp.inception.active.learning.sidebar.ActiveLearningSidebarFactory.create(ActiveLearningSidebarFactory.java:52) ~[inception-active-learning-0.4.0-SNAPSHOT.jar!/:?]
	at de.tudarmstadt.ukp.clarin.webanno.ui.annotation.sidebar.SidebarPanel$1.getPanel(SidebarPanel.java:101) ~[webanno-ui-annotation-3.5.0-SNAPSHOT.jar!/:?]
	at de.tudarmstadt.ukp.clarin.webanno.ui.annotation.sidebar.SidebarPanel$1.getPanel(SidebarPanel.java:89) ~[webanno-ui-annotation-3.5.0-SNAPSHOT.jar!/:?]
	at org.apache.wicket.extensions.markup.html.tabs.TabbedPanel.setCurrentTab(TabbedPanel.java:422) ~[wicket-extensions-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.extensions.markup.html.tabs.TabbedPanel.setSelectedTab(TabbedPanel.java:397) ~[wicket-extensions-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.extensions.markup.html.tabs.TabbedPanel.onBeforeRender(TabbedPanel.java:268) ~[wicket-extensions-7.10.0.jar!/:7.10.0]
	at de.tudarmstadt.ukp.clarin.webanno.ui.annotation.sidebar.SidebarTabbedPanel.onBeforeRender(SidebarTabbedPanel.java:82) ~[webanno-ui-annotation-3.5.0-SNAPSHOT.jar!/:?]
	at org.apache.wicket.Component.internalBeforeRender(Component.java:950) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.Component.beforeRender(Component.java:1018) ~[wicket-core-7.10.0.jar!/:7.10.0]
	at org.apache.wicket.MarkupContainer.onBeforeRenderChildren(MarkupContainer.java:1825) ~[wicket-core-7.10.0.jar!/:7.10.0]
	... 150 more
```

**Please complete the following information:**
 - INCEpTION 𝜶 -- 0.4.0-SNAPSHOT (2018-07-02 13:20:46, build 5bedb5eb63bf9a8a2771fc2f56c0b0bf09a54cc7)
 - chrome

### PR DESCRIPTION
**Pull request by matchlesswei**
If a project doesn't have recommenders, no layer can be selected in the active learning sidebar. In this case, active learning sidebar shouldn't return a `featureEditor`. Instead, it returns an empty invisible label.

Fixes #370.

### CODE REVIEW

File: inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java [hunk 1]
```diff
@@ -440,14 +442,11 @@ private void jumpToRecommendationLocationAndHighlightRecommendation(AjaxRequestT
...
-                .getFeature(currentRecommendation.getFeature(), selectedLayer.getObject());
-        }
-        else {
-            annotationFeature = annotationService
-                .listAnnotationFeature(selectedLayer.getObject()).get(0);
-        }
+        annotationFeature = annotationService
+            .getFeature(currentRecommendation.getFeature(), selectedLayer.getObject());
+
+        annotationFeature = annotationService.listAnnotationFeature(selectedLayer.getObject())
```
Comment (reckart) [inline inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java]: Why are you assigning annotationFeature twice in a row?

Comment (matchlesswei) [reply inception-active-learning/src/main/java/de/tudarmstadt/ukp/inception/active/learning/sidebar/ActiveLearningSidebar.java]: Sorry... It's a mistake caused by changing the old codes.

**Comment by matchlesswei** [comment]
Yes, I can add the message and set the `sessionControlForm` as invisible if there are no layers with recommenders. Do we create a new issue for it, or change in this PR? The current internal error happens even when the user doesn't open the active learning sidebar.

**Comment by reckart** [comment]
@matchlesswei You cannot really create a feature editor once the user has selected a feature  normally only when the user has actually started the session. Before that, you have no idea which editor to create - that is basically the reason why the error occurs. The reason the error happens even if the sidebar is not opened is that the AL sidebar is the first sidebar and thus by default the active one. 

I'd say its ok to include the message in this issue/PR.

**Comment by reckart** [comment]
Fixes #370.

**Pull request merged**